### PR TITLE
fix(frontend): Refactor frontend config

### PR DESCRIPTION
**Pull request by amanape**
**End-user friendly description of the problem this fixes or functionality that this introduces**

- [ ] Include this change in the Release Notes. If checked, you must provide an **end-user friendly** description for your change below

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

- No more hardcoded `localhost:3000` for the server base url. Instead, we can set it in `.env` (fallback set to "localhost:3000")
- Move `OpenHands` API types to `open-hands.types.ts`
- Migrate all remaining API into `OpenHands` API class
- Created an `open-hands.utils.ts` file for better handling of return types
- Removed old logic in `vite.config.ts` relating to a proxy server

---
**Link of any specific issues this addresses**

### CODE REVIEW

**Comment by tobitege** [comment]
What happens if there's no .env file containing the baseUrl? 🤔

**Comment by tobitege** [comment]
Re: tests
"55 skipped" is that normal (with 51 run)? I guess "wip" status?

Ignorable?
https://github.com/All-Hands-AI/OpenHands/actions/runs/11232337919/job/31223725707?pr=4261#step:5:37
```
stderr | __tests__/services/session.test.ts > startNewSession > Should start a new session with the current settings
[MSW] Warning: intercepted a request without a matching request handler:
```

**Comment by amanape** [comment]
@tobitege Added a fallback if the env was not set. And for tests, yes, I ignored a lot of them making changes to the new UI so I'll update them later

File: frontend/.env.sample [hunk 1]
```diff
@@ -1,7 +1,4 @@
-VITE_BACKEND_HOST="127.0.0.1:3000"
-VITE_USE_TLS="false"
-VITE_INSECURE_SKIP_VERIFY="false"
-VITE_FRONTEND_PORT="3001"
+VITE_BACKEND_BASE_URL="localhost:3000"
```
Comment (tobitege) [inline frontend/.env.sample:1]: Would it make sense to add a comment like `# Base URL without protocol` or similar?

Comment (amanape) [reply frontend/.env.sample]: Done

**Review by rbren — APPROVED** [review]

**Pull request merged**